TKeazirian.HTTPServer.Tests/ControllerTests.cs

```diff
@@ -5,37 +5,54 @@ namespace TKeazirian.HTTPServer.Tests;
 
 public class ControllerTests
 {
-    private const string NewLine = "\r\n";
-
     [Fact]
     public void EchoRequestBodyFormatsResponse()
     {
         Controller controller = new Controller();
 
         string testRequest = HelperFunctions.FormatTestRequest("POST", "/test_path", "Hello, World!");
 
-        string expectedResponse =
-            $"HTTP/1.1 200 OK{NewLine}" +
-            $"Content-Type: plain/text{NewLine}" +
-            $"Content-Length:13{NewLine}{NewLine}" +
-            $"Hello, World!";
+        string statusCode = Constants.Status200;
+        string headers = Parser.ParseHeaders(testRequest);
+        string? body = Parser.ParseRequestBody(testRequest);
+
+        string expectedResponse = HelperFunctions.FormatTestResponseWithHeaders(statusCode, headers, body);
 
         var actualResponse = controller.EchoRequestBody(testRequest);
 
         Assert.Equal(expectedResponse, actualResponse);
     }
 
+    [Fact]
+    public void CreateBodyToSendForGetRequest()
```

Comment:
This test description makes me think that what's being tested is a function that's responsible for creating the response body, but it looks like this is testing more than that. Can you update it to be more descriptive?